argus/backend/service/client_service.py

```diff
@@ -109,12 +110,19 @@ def submit_results(self, run_type: str, run_id: str, results: dict) -> dict[str,
                                                            cells=cells, run_id=run_id)
         table_name = results["meta"]["name"]
         sut_timestamp = results["sut_timestamp"]
+        result_failed = False
         for cell in cells:
             cell.update_cell_status_based_on_rules(table_metadata, best_results)
+            if cell.status == "ERROR":
+                result_failed = True
             ArgusGenericResultData(test_id=run.test_id,
                                    run_id=run.id,
                                    name=table_name,
                                    sut_timestamp=sut_timestamp,
                                    **asdict(cell)
                                    ).save()
-        return {"status": "ok", "message": "Results submitted"}
+        if result_failed:
+            raise DataValidationError()
+        else:
```

Comment:
Just to clean this up, you can now remove the `else:` and just leave `return {...}` statement

Reply:
done